### PR TITLE
[SYCL][ESIMD] Replace "dim", "dimensions" with "size", "sizes", etc.

### PR DESCRIPTION
**Pull request by vasilytric**
There is no changes in the tests logic.
There are just changes that are related to replacing "dim" with "size".
Also older `get_dimensions` and `get_all_dimensions` are not deleted yet, they saved for the backward compatibility and will be deleted later.

### CODE REVIEW

File: SYCL/ESIMD/api/functional/type_coverage.hpp [hunk 1]
```diff
@@ -282,13 +282,23 @@ template <tested_types required> auto get_tested_types() {
...
+
+// It's a deprecated function and it exists only for backward compatibility and
+// it should be deleted in the future.
+auto inline get_all_dimensions() { return get_all_sizes(); }
+
+// It's a deprecated function and it exists only for backward compatibility and
+// it should be deleted in the future.
+template <int... Values> auto inline get_dimensions() {
+  return integer_pack<Values...>::generate_unnamed();
+}
```
Comment (yuriykoch) [inline SYCL/ESIMD/api/functional/type_coverage.hpp]:
```suggestion
// It's a deprecated function and it exists only for backward compatibility and
// it should be deleted in the future. Use get_sizes() instead
template <int... Values> auto inline get_dimensions() {
  return get_sizes<Values...>();
}
```

File: SYCL/ESIMD/api/functional/type_coverage.hpp [hunk 2]
```diff
@@ -282,13 +282,23 @@ template <tested_types required> auto get_tested_types() {
...
 
 // Factory method to retrieve pre-defined values_pack, to have the same
-// default dimensions over the tests
-auto inline get_all_dimensions() { return get_dimensions<1, 8, 16, 32>(); }
+// default sizes over the tests
+auto inline get_all_sizes() { return get_sizes<1, 8, 16, 32>(); }
+
+// It's a deprecated function and it exists only for backward compatibility and
+// it should be deleted in the future.
+auto inline get_all_dimensions() { return get_all_sizes(); }
```
Comment (yuriykoch) [inline SYCL/ESIMD/api/functional/type_coverage.hpp]:
```suggestion
// It's a deprecated function and it exists only for backward compatibility and
// it should be deleted in the future. Use get_all_sizes() instead
auto inline get_all_dimensions() { return get_all_sizes(); }
```

Comment (vasilytric) [reply SYCL/ESIMD/api/functional/type_coverage.hpp]: I have applied this comment in this commit: https://github.com/intel/llvm-test-suite/pull/803/commits/f9168ed3dc8b995f5ed4f7f35384cb2d9719bfb9

File: SYCL/ESIMD/api/functional/type_coverage.hpp (hunk 1 above)
Comment (vasilytric) [reply SYCL/ESIMD/api/functional/type_coverage.hpp]: I have applied this comment in this commit: https://github.com/intel/llvm-test-suite/pull/803/commits/f9168ed3dc8b995f5ed4f7f35384cb2d9719bfb9

**Comment by vasilytric** [comment]
@sndmitriev Would you mind to take a look?

**Review by sndmitriev — DISMISSED** [review]
LGTM

**Comment by vasilytric** [comment]
@kbobrovs Sergey have approved this PR would you mind to merge this PR?

**Review by sndmitriev — APPROVED** [review]

**Comment by vasilytric** [comment]
@kbobrovs Would you mind to merge this PR?

**Pull request merged**